librz/core/cprint.c

```diff
@@ -30,4 +30,234 @@ RZ_API char *rz_core_print_string_c_cpp(RzCore *core) {
 	}
 	rz_strbuf_append(sb, "\";");
 	return rz_strbuf_drain(sb);
-}
\ No newline at end of file
+}
+
+/**
+ * \brief Print hexdump diff between \p aa and \p ba with \p len
+ */
+RZ_API bool rz_core_print_cmp(RZ_NONNULL RzCore *core, ut64 aa, ut64 ba, ut64 len) {
+	rz_return_val_if_fail(core && core->cons && len > 0, false);
+	ut8 *a = malloc(len);
+	if (!a) {
+		return false;
+	}
+	ut8 *b = malloc(len);
+	if (!b) {
+		free(a);
+		return false;
+	}
+
+	RZ_LOG_VERBOSE("diff 0x%" PFMT64x " 0x%" PFMT64x " with len:%" PFMT64d "\n", aa, ba, len);
+
+	rz_io_read_at(core->io, aa, a, (int)len);
+	rz_io_read_at(core->io, ba, b, (int)len);
+	int col = core->cons->columns > 123;
+	rz_print_hexdiff(core->print, aa, a,
+		ba, b, (int)len, col);
+	free(a);
+	free(b);
+	return true;
+}
+
+static inline st8 format_type_to_base(const RzCorePrintFormatType format, const ut8 n) {
+	static const st8 bases[][9] = {
+		{ 0, 8 },
+		{ 0, -1, -10, [4] = 10, [8] = -8 },
+		{ 0, 16, 32, [4] = 32, [8] = 64 },
+	};
+	if (format >= RZ_CORE_PRINT_FORMAT_TYPE_INVALID || n >= sizeof(bases[0])) {
+		return 0;
+	}
+	return bases[format][n];
+}
+
+static inline void fix_size_from_format(const RzCorePrintFormatType format, ut8 *size) {
+	if (format != RZ_CORE_PRINT_FORMAT_TYPE_INTEGER) {
+		return;
+	}
+	static const st8 sizes[] = {
+		0, 4, 2, [4] = 4, [8] = 4
+	};
+	if (*size >= sizeof(sizes)) {
+		return;
+	}
+	*size = sizes[*size];
+}
+
+static inline void len_fixup(RzCore *core, ut64 *addr, int *len) {
+	if (!len || *len >= 0) {
+		return;
+	}
+	*len = -*len;
+	if (*len > core->blocksize_max) {
+		RZ_LOG_ERROR("this block size is too big (0x%" PFMT32x
+			     " < 0x%" PFMT32x ").",
+			*len,
+			core->blocksize_max);
+		*len = (int)core->blocksize_max;
+	}
+	if (addr) {
+		*addr = *addr - *len;
+	}
+}
+
+/**
+ * \brief Print dump at \p addr
+ * \param n Word size by bytes (1,2,4,8)
+ * \param len Dump bytes length
+ * \param format Print format, such as RZ_CORE_PRINT_FORMAT_TYPE_HEXADECIMAL
+ */
+RZ_API bool rz_core_print_dump(RZ_NONNULL RzCore *core, RZ_NULLABLE RzCmdStateOutput *state,
```

Comment:
in terms of output, isn't this supposed to be the same as `rz_core_print_hexdump`?

Reply:
I actually don't understand why we have an API like the `rz_core_print_hexdump_or_hexdiff`, but just because it existed in the past, so I made it an API

Reply:
You are absolutely right, if it's a printing function you can remove an RZ_API mark. That's what I usually do in my conversion PRs.